packages/element-lib/src/sidetree-v2/protocol/sync.js

```diff
@@ -0,0 +1,27 @@
+const {
+  executeSequentially,
+  syncTransaction,
+  isTransactionValid,
+} = require('../func');
+
+const sync = sidetree => async () => {
+  // Get a set of transactions that have only been processed from cache
+  const transactionsAlreadyProcessed = await sidetree.db.readCollection('transaction');
+  const processedSet = new Set(transactionsAlreadyProcessed.map(t => t.transactionNumber));
+  // Get all transactions from the smart contract
+  const transactions = await sidetree.blockchain.getTransactions(
```

Comment:
This looks very expensive. I think ideally the node should always store transactions, there is no reason we should be getting 100% of all transactions ever created from ethereum every-time sync is called.

Reply:
Yes! Let's merge this and I will do that in a separate PR